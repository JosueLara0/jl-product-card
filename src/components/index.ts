//? first pattern

//* components
import { ProductCard as ProductCardHOC } from "./ProductCard";
import { ProductButtons } from "./ProductButtons";
import { ProductImage } from "./ProductImage";
import { ProductTitle } from "./ProductTitle";
//* interfaces
import { ProductCardHOCProps } from "../interfaces/interfaces";

export const ProductCard: ProductCardHOCProps = Object.assign(ProductCardHOC, {
  Title: ProductTitle,
  Image: ProductImage,
  Buttons: ProductButtons,
});

//? second pattern

//* components
export * from "./ProductCard";
export { ProductButtons } from "./ProductButtons";
export { ProductTitle } from "./ProductTitle";
export { ProductImage } from "./ProductImage";
