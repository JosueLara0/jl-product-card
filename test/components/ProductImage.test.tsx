//* libraries
import React from 'react';
import renderer from 'react-test-renderer';
//* components
import { ProductImage, ProductCard } from '../../src/components';
//* data
import { product2 } from '../data/products';

describe('Tests in ProductImage component', () => {
  test('should render component with the custom img', () => {
    const wrapper = renderer.create(<ProductImage img="./coffe-mug.png" />);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should render component with the product img', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
