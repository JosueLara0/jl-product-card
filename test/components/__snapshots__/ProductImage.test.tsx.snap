// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in ProductImage component should render component with the custom img 1`] = `
<img
  alt="Product img"
  className="productImg undefined"
  src="./coffe-mug.png"
/>
`;

exports[`Tests in ProductImage component should render component with the product img 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product img"
    className="productImg undefined"
    src="./coffe-mug.png"
  />
</div>
`;
